File: modules/live-activities/src/LiveActivitiesModule.ts
import { NativeModule, requireNativeModule } from 'expo';

import type {
  ActiveActivity,
  LiveActivitiesModuleEvents,
} from './LiveActivities.types';

declare class LiveActivitiesModule extends NativeModule<LiveActivitiesModuleEvents> {
  areActivitiesEnabled(): boolean;
  getPushToStartToken(): string | null;
  getActiveActivities(): ActiveActivity[];
  startObservingTokens(): void;
}

export default requireNativeModule<LiveActivitiesModule>('LiveActivities');

File: modules/live-activities/android/src/main/java/expo/modules/liveactivities/LiveActivitiesModule.kt
package expo.modules.liveactivities

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

/** Stub: Live Activities são iOS-only neste tutorial. */
class LiveActivitiesModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("LiveActivities")

    Events("pushToStartTokenDidChange", "pushToUpdateTokenDidChange")

    Function("areActivitiesEnabled") { false }

    Function("getPushToStartToken") { null }

    Function("getActiveActivities") {
      emptyList<Map<String, Any>>()
    }

    Function("startObservingTokens") { }
  }
}
